components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/util/ResourceUtils.java

```diff
@@ -0,0 +1,110 @@
+import org.wso2.carbon.identity.tenant.resource.manager.exception.TenantResourceManagementServerException;
+import org.wso2.carbon.identity.tenant.resource.manager.internal.TenantResourceManagerDataHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility methods for tenant resource management.
+ */
+public class ResourceUtils {
+
+    private static final Log log = LogFactory.getLog(ResourceUtils.class);
+
+    /**
+     * This method can be used to generate a TenantResourceManagementServerException from
+     * TenantResourceConstants.ErrorMessages object when no exception is thrown.
+     *
+     * @param error TenantResourceConstants.ErrorMessages.
+     * @param data  data to replace if message needs to be replaced.
+     * @return TenantResourceManagementServerException.
+     */
+    public static TenantResourceManagementServerException handleServerException(
+            TenantResourceConstants.ErrorMessages error, String... data) {
+
+        String message = populateMessageWithData(error, data);
+        return new TenantResourceManagementServerException(message, error.getCode());
+    }
+
+    public static TenantResourceManagementServerException handleServerException(
+            TenantResourceConstants.ErrorMessages error, Throwable e, String... data) {
+
+        String message = populateMessageWithData(error, data);
+        return new TenantResourceManagementServerException(message, error.getCode(), e);
+    }
+
+    /**
+     * This method can be used to get publisher configuration file from the configuration store.
+     *
+     * @param eventPublisherName Event Publisher Name.
+     * @return event publisher file
+     * @throws ConfigurationManagementException
+     */
+    public static ResourceFile getResourceFile(String eventPublisherName) throws ConfigurationManagementException {
+
+        List<ResourceFile> fileList = new ArrayList<>();
+        Resources resources = TenantResourceManagerDataHolder.getInstance().getConfigurationManager()
+                .getResourcesByType(TenantResourceConstants.PUBLISHER);
+        for (Resource resource : resources.getResources()) {
+            if (eventPublisherName.equals(resource.getResourceName())) {
+                fileList = TenantResourceManagerDataHolder.getInstance().getConfigurationManager()
+                        .getFiles(TenantResourceConstants.PUBLISHER, eventPublisherName);
+                break;
+            }
+        }
+
+        if (fileList.size() > 1) {
+            if (log.isDebugEnabled()) {
+                log.debug("More then one file with the name: " + eventPublisherName);
+            }
+            return null;
+        } else if (fileList.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("No file with the name: " + eventPublisherName);
+            }
+            return null;
+        } else {
+            return fileList.get(0);
+        }
+    }
```

Comment:
I don't think this method should come to a utility.
What holds you from having this in the service implementation